packages/std/src/lib.rs

```diff
@@ -1,15 +1,40 @@
-// these two are conditionally compiled, only for wasm32
-pub mod exports;
-pub mod imports;
-
-pub mod encoding;
-pub mod errors;
-pub mod memory;
-pub mod mock;
-pub mod serde;
-pub mod storage;
-pub mod traits;
-pub mod types;
-
-// not exposed
+// Exposed on all platforms
+
+mod encoding;
+mod errors;
+mod mock;
+mod serde;
+mod storage;
+mod traits;
+mod types;
+
+pub use crate::encoding::Binary;
+pub use crate::errors::{
```

Comment:
I would make errors `pub mod` as well. There are lots of symbols there. Maybe this is enough, but I don't know

Reply:
While it works well for now, I'm not convinced that sorting by types (like errors, traits, types) is a good strategy long term. It is a centralized aproach where one file needs to know everything that is going on in the entire crate, creating a lot of corss-module dependencies. We are already seeing things like `pub struct Extern<S: Storage, A: Api>` in `traits.rs`.

Instead, I think we should group by domain. E.g. `encoding` can have traits, types, implementations and errors that are specific to the encoding functionality.

I don't the need to move around symbols at the moment, but being able to do so without breaking the public API is valuable I think.

Reply:
I agree. Let's do as you suggested